core/src/main/java/io/aiven/klaw/helpers/db/rdbms/SelectDataJdbc.java

```diff
@@ -1169,4 +1156,57 @@ public int findAllComponentsCountForTeam(Integer teamId, int tenantId) {
   public int getAllTopicsCountInAllTenants() {
     return ((Long) topicRepo.findAllTopicsCount().get(0)[0]).intValue();
   }
+
+  // teamId is requestedBy. For 'topics' all the requests are assigned to the same team, except
+  // claim requests
+  public Map<String, Map<String, Long>> getTopicRequestsCounts(
+      int teamId, RequestMode requestMode, int tenantId) {
+    Map<String, Map<String, Long>> allCountsMap = new HashMap<>();
+
+    Map<String, Long> operationTypeCountsMap = new HashMap<>();
+    Map<String, Long> statusCountsMap = new HashMap<>();
+
+    if (RequestMode.MY_REQUESTS == requestMode) {
+      List<Object[]> topicRequestsOperationTypObj =
+          topicRequestsRepo.findAllTopicRequestsGroupByOperationType(teamId, tenantId);
+      topicRequestsOperationTypObj.forEach(
+          topicReqObjs ->
+              operationTypeCountsMap.put((String) topicReqObjs[0], (Long) topicReqObjs[1]));
+
+      List<Object[]> topicRequestsStatusObj =
+          topicRequestsRepo.findAllTopicRequestsGroupByStatus(teamId, tenantId);
+      topicRequestsStatusObj.forEach(
+          topicReqObjs -> statusCountsMap.put((String) topicReqObjs[0], (Long) topicReqObjs[1]));
+    } else if (RequestMode.TO_APPROVE == requestMode) {
+      List<Object[]> topicRequestsStatusObj =
+          topicRequestsRepo.findAllTopicRequestsGroupByStatus(teamId, tenantId);
+      topicRequestsStatusObj.forEach(
+          topicReqObjs -> statusCountsMap.put((String) topicReqObjs[0], (Long) topicReqObjs[1]));
+
+      long assignedToClaimReqs =
+          topicRequestsRepo.countAllTopicRequestsByDescriptionAndTopictype(
+              tenantId, "" + teamId, RequestOperationType.CLAIM.value);
+      List<Object[]> topicRequestsOperationTypObj =
+          topicRequestsRepo.findAllTopicRequestsGroupByOperationType(teamId, tenantId);
+      topicRequestsOperationTypObj.forEach(
+          topicReqObjs ->
+              operationTypeCountsMap.put((String) topicReqObjs[0], (Long) topicReqObjs[1]));
+
+      operationTypeCountsMap.put(RequestOperationType.CLAIM.value, assignedToClaimReqs);
+    }
+
+    for (RequestStatus requestStatus : RequestStatus.values()) {
+      if (!statusCountsMap.containsKey(requestStatus.value))
+        statusCountsMap.put(requestStatus.value, 0L);
+    }
+    for (RequestOperationType requestOperationType : RequestOperationType.values()) {
+      if (!operationTypeCountsMap.containsKey(requestOperationType.value))
+        operationTypeCountsMap.put(requestOperationType.value, 0L);
+    }
+
```

Comment:
NitPick: You could do something like below which could be used by both RequestOperationType and Request Stats.

  public void addStatisticIfMissing(Map<String,Long> stats,String statKey){
    if(!stats.containsKey(statKey)) {
      stats.put(statKey,0L);
    }
  }

Reply:
ignored as there are just 2 tiny blocks